frontend/src/stores/search.ts

```diff
@@ -223,35 +203,32 @@ export const useSearchStore = defineStore("search", {
       return this.getSearchPath()
     },
     /**
-     * Returns localized search path for the given search type.
+     * Returns localized frontend search path for the given search type.
      *
      * If search type is not provided, returns the path for the current search type.
      * If query is not provided, returns current query parameters.
      */
     getSearchPath({
       type,
       query,
-    }: { type?: SearchType; query?: ApiQueryParams } = {}): string {
+    }: { type?: SearchType; query?: PaginatedSearchQuery } = {}): string {
       const searchType = type || this.searchType
-      let queryParams
-      if (!query) {
-        if (type && isSearchTypeSupported(type)) {
-          queryParams = computeQueryParams(
-            type,
-            this.filters,
-            this.searchTerm,
-            "frontend"
-          )
-        } else {
-          queryParams = this.frontendSearchUrlParams
-        }
-      } else {
-        queryParams = query
+      let queryParams = query
+      if (queryParams === undefined) {
+        queryParams =
+          searchType === undefined
+            ? this.frontendSearchUrlParams
+            : computeQueryParams(
+                searchType,
+                this.filters,
+                this.searchTerm,
+                "frontend"
+              )
```

Comment:
What's the difference between using `frontendSearchUrlParams` and calling `computeQueryParams` with the frontend strategy? `frontendSearchUrlParams`'s implementation looks the same as the else branch of this ternary. Is the difference in passing `searchType` directly vs using `this.searchType` as `frontendSearchUrlParams` does? In what circumstances would `type` be undefined? Can those circumstances safely fallback to `ALL_MEDIA`, maybe?

If anything, a comment here would be helpful to explain why the branches are different and why an undefined search type dictates which we choose. Whatever the distinction is between these two branches seems to be very subtle and is completely lost on me.

Reply:
I simplified this code. The `type` is passed here when creating a search page link (in the content switcher or in the  content link on the All content search page) for a type that's different from the current search type.